cmake/Corrosion.cmake

```diff
@@ -1086,8 +1085,29 @@ function(corrosion_link_libraries target_name)
             $<TARGET_PROPERTY:${library},LINKER_LANGUAGE>
         )
 
-        corrosion_add_target_local_rustflags(${target_name} "-L$<TARGET_LINKER_FILE_DIR:${library}>")
-        corrosion_add_target_local_rustflags(${target_name} "-l$<TARGET_LINKER_FILE_BASE_NAME:${library}>")
+        if (TARGET ${library})
+            corrosion_add_target_local_rustflags(${target_name}
+                "-L$<TARGET_LINKER_FILE_DIR:${library}>"
+                "-l$<TARGET_LINKER_FILE_BASE_NAME:${library}>"
+            )
+            add_dependencies(_cargo-build_${target_name} ${library})
+            get_property(libs TARGET ${library} PROPERTY INTERFACE_LINK_LIBRARIES)
+            corrosion_link_libraries(${target_name} ${libs})
+        elseif(IS_ABSOLUTE ${library})
+            # Linking via full path
+            corrosion_add_target_local_rustflags(${target_name} "-Clink-arg=${library}")
```

Comment:
Could you add 
- a link to the relavant rustc documentatation (https://doc.rust-lang.org/rustc/command-line-arguments.html#linking-modifiers-verbatim)
- ideally a test project that utilizes absolute path linking.